### PR TITLE
fix: trying to access to context on a side effect

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
References #722

### CODE REVIEW

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala [hunk 1]
```diff
@@ -131,9 +131,22 @@ abstract class EventSourcedEntityRouter[S, E <: EventSourcedEntity[S]](protected
         val snapshot =
           if (shouldSnapshot) Option(endState)
           else None
-        CommandResult(events.toVector, commandEffect.secondaryEffect(endState), snapshot, currentSequence)
+
+        try {
+          context.asInstanceOf[CommandContextImpl].checkActive()
```
Comment (franciscolopezsancho) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala:136]: As this is the `EventSourcedEntityRouter` we know it has been created with an instance of `CommandContextImpl`. Not too sure though if this is the right approach.

**Comment by franciscolopezsancho** [comment]
Even though the integration test passes and the application also works when `sbt run` it raises the same exception (#722) when running an unit test. I understand that the test kit is not leveraging `EventSourceEntityImpl.runEntity` which is what is fixed here.

**Comment by johanandren** [comment]
The testkit is a little bit tricky because we expect that the secondary effect can be evaluated lazily in the `EventSourcedResultImpl` where we do not have access to the actual entity instance, but because of this issue I think we'll have to do that eagerly in the generated event sourced testkit, so that we can surround it with setting and clearing out the context, and then pass the evaluated effects to EventSourcedResultImpl.

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala (hunk 1 above)
Comment (johanandren) [reply sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala:136]: We shouldn't check active though but rather toggle it to active, no such method right now (but only add that if needed and you see it fail without this `checkActive()` call, I think we may be fine without it)

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala [hunk 2]
```diff
@@ -131,9 +131,22 @@ abstract class EventSourcedEntityRouter[S, E <: EventSourcedEntity[S]](protected
...
+          entity._internalSetCommandContext(Optional.of(context))
```
Comment (johanandren) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala:137]:
```suggestion
          // side effect callbacks may want to access context or components which is valid
          entity._internalSetCommandContext(Optional.of(context))
```

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala [hunk 3]
```diff
@@ -252,20 +251,21 @@ final class EventSourcedEntitiesImpl(
...
-
   private class EventSourcedEntityContextImpl(override final val entityId: String)
       extends AbstractContext(system)
       with EventSourcedEntityContext
   private final class EventContextImpl(entityId: String, override val sequenceNumber: Long)
       extends EventSourcedEntityContextImpl(entityId)
       with EventContext
 }
+
+class CommandContextImpl(
```
Comment (johanandren) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala:262]: `private[eventsourcedentity]` and `final` (if it still needs to be out here)

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala [hunk 4]
```diff
@@ -131,9 +131,23 @@ abstract class EventSourcedEntityRouter[S, E <: EventSourcedEntity[S]](protected
...
+          // side effect callbacks may want to access context or components which is valid
+          entity._internalSetCommandContext(Optional.of(context))
+          CommandResult(events.toVector, commandEffect.secondaryEffect(endState), snapshot, currentSequence)
+        } finally {
+          entity._internalSetCommandContext(Optional.empty())
+        }
       case NoPrimaryEffect =>
-        CommandResult(Vector.empty, commandEffect.secondaryEffect(_stateOrEmpty()), None, context.sequenceNumber())
+        try {
+          context.asInstanceOf[CommandContextImpl].checkActive()
```
Comment (johanandren) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntityRouter.scala:145]: Same here
```suggestion
```

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala (hunk 3 above)
Comment (franciscolopezsancho) [reply sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala:262]: not anymore, I'll put it back

**Comment by franciscolopezsancho** [comment]
continuing in #737

**Pull request closed without merging**